src/main/java/org/spongepowered/mod/SpongeMod.java

```diff
@@ -246,4 +260,42 @@ public String getId() {
     public Object getInstance() {
         return getMod();
     }
+
+    public SpongeGameRegistry getSpongeRegistry() {
+        return this.registry;
+    }
+
+    public void registerAllEnabledWorlds() {
+        File[] directoryListing = DimensionManager.getCurrentSaveRootDirectory().listFiles();
+        if (directoryListing != null) {
+            for (File child : directoryListing) {
+                if (child.isDirectory()) {
+                    File levelData = new File(child, "level_sponge.dat");
+                    if (levelData.exists()) {
+                        try {
+                            NBTTagCompound nbt = CompressedStreamTools.readCompressed(new FileInputStream(levelData));
+                            if (nbt.hasKey(getModId())) {
+                                NBTTagCompound spongeData = nbt.getCompoundTag(getModId());
+                                if (spongeData.hasKey("dimensionId") && spongeData.getBoolean("enabled")) {
+                                    for (Map.Entry<Class<? extends Dimension>, DimensionType> mapEntry : getSpongeRegistry().dimensionClassMappings
+                                            .entrySet()) {
+                                        if (mapEntry.getKey().getCanonicalName().equalsIgnoreCase(spongeData.getString("dimensionType"))) {
+                                            if (!DimensionManager.isDimensionRegistered(spongeData.getInteger("dimensionId"))) {
+                                                DimensionManager.registerDimension(spongeData.getInteger("dimensionId"),
+                                                        ((SpongeDimensionType) mapEntry.getValue()).getDimensionTypeId());
```

Comment:
Way too much nesting. If this happens normally I would invert the if statements and keep the indentation level low.
e.g. 
```java
if (directoryListing == null) {
    return;
}
for (File child : directoryListing) {
    if (!child.isDirectory()) {
        continue;
    }
    [...]
}
```